### DISCUSSION TITLE
Cross Compilation for arm Architecture

### DISCUSSION THREAD

**Discussion by harshakhmk** [milvus-io/milvus#29325] [Q&A and General discussion]
Does Milvus support cross compilation for arm Architecture , if yes can I get complete guide of steps to run build and start the server on arm Architecture

Any leads on this would be highly appreciated

**Comment by liliu-z** [comment]
Milvus will release specific ARM image, but that is for host with 4k page size for now. Others will be supported soon

**Reply by harshakhmk** [reply]
But is cross Compilation possible for arm in Milvus?

**Reply by liliu-z** [reply]
Yes, just need to make sure the host you use to compile is also an ARM host

**Reply by harshakhmk** [reply]
Where can I find all the steps for cross compilation for arm Architecture

**Reply by presburger** [reply]
@harshakhmk Cross-compiling Milvus can be a highly challenging process due to its complex dependencies. Currently, for the compilation of ARM64, we complete the process natively in an ARM64 Linux environment.

**Reply by harshakhmk** [reply]
For building arm64 host machine, 
```
# Clone github repository.
$ git clone https://github.com/milvus-io/milvus.git

# Install third-party dependencies.
$ cd milvus/
$ ./scripts/install_deps.sh

# Compile Milvus.
$ make
```
Does this steps work or can you provide any specific steps to be followed for arm64 host machine

**Reply by harshakhmk** [reply]
Kindly confirm the above steps for Cross compilation or if anything needed to be added
Any opinion on the above, would be highly appreciated

**Reply by yhmo** [reply]
@harshakhmk 
Our engineers have fixed the compile error for ARM64/AArch64. You can download the latest source code of master branch to compile it on ARM64/AArch64 host.

Yes the steps are correct:
```
# Clone github repository.
$ git clone https://github.com/milvus-io/milvus.git

# Install third-party dependencies.
$ cd milvus/
$ ./scripts/install_deps.sh

# Compile Milvus.
$ make
```
